source/common/upstream/subset_lb.cc

```diff
@@ -281,6 +333,22 @@ HostConstSharedPtr SubsetLoadBalancer::tryChooseHostFromContext(LoadBalancerCont
     return nullptr;
   }
 
+  if (single_) {
+    for (const auto& entry : match_criteria->metadataMatchCriteria()) {
+      if (entry->name() == single_key_) {
```

Comment:
Should we only match if there is only a single criteria listed? Seems a bit odd to have a long list of criteria only to have most of them ignored

Reply:
I would be fine with that. I'm not familiar enough with metadata matches and how they're generally used to know which behavior is better. 

What would be the behavior if multiple metadataMatchCritieria are provided here? Just go straight to fallback?

Reply:
Thinking through it, I could potentially imagine someone adding multiple criteria, then going through router to select a cluster, and different clusters matching on different metadata, so I'm leaning towards leaving this as-is.